mitodl/mit-open
# ETL Pipelines for MITx Online, MIT edX, Open Learning Library

# What are the relevant tickets?
Closes #125 
Closes #126 
Closes #127

# Description (What does it do?)
- Migrates the following ETL pipelines from `course_catalog` to `learning_resources`:
  - MITx Online 
  - Open Learning Library
  -  MITx courses on edX
- Consistently use `etl_source` instead of `platform` where applicable as inputs to various ETL functions
- Moves the `etl_source` values to an Enum class
- Adds back scheduled ETL tasks for the above to celerybeat.

# How can this be tested?
You will need to copy a bunch of settings into your .env file from open-discussions heroku or as listed here:

```
EDX_API_URL
EDX_API_CLIENT_ID
EDX_API_CLIENT_SECRET
EDX_API_ACCESS_TOKEN_UR
EDX_LEARNING_COURSE_BUCKET_NAME

MITX_ONLINE_BASE_URL=https://mitxonline.mit.edu/
MITX_ONLINE_COURSES_API_URL=https://mitxonline.mit.edu/api/courses/
MITX_ONLINE_PROGRAMS_API_URL=https://mitxonline.mit.edu/api/programs/
MITX_ONLINE_LEARNING_COURSE_BUCKET_NAME=mitx-etl-mitxonline-production

OLL_API_URL
OLL_API_ACCESS_TOKEN_URL
OLL_API_CLIENT_ID
OLL_API_CLIENT_SECRET
OLL_BASE_URL
OLL_ALT_URL
```

- Run `./manage.py update_platforms` to get the latest `Platform` objects created
- Run `./manage.py backpopulate_mitxonline_data`
- Run `./manage.py backpopulate_mit_edx_data`
- Run `./manage.py backpopulate_oll_data`

After running the above, you should have published courses and runs from each source in the database (can check via API or django admin).

These two may take some time, you can interrupt them after a while:
- Run `./manage.py backpopulate_mitxonline_files`
- Run `./manage.py backpopulate_mit_edx_files`

Afterward, you should have `ContentFile` objects associated with MITX Online and MIT EdX course runs.

# Additional Context
